File: NativeAPI/Python/b2_download_file_by_name.py
import requests

download_url = ""  # Provided by b2_authorize_account
account_authorization_token = ""  # Provided by b2_authorize_account
bucket_name = ""  # The name of the bucket containing the file
file_name = ""  # The name of the file you want to download

response = requests.get(
    f"{download_url}/file/{bucket_name}/{file_name}",
    headers={"Authorization": account_authorization_token},
)
content = response.text
print(content)

File: NativeAPI/Java/B2DownloadFileByName.java
package com.backblaze.examples;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class B2DownloadFileByName {
    public static void main(String[] args) {
        String downloadUrl = ""; // Provided by b2_authorize_account
        String accountAuthorizationToken = ""; // Provided by b2_authorize_account
        String bucketName = ""; // The bucket name where the file exists
        String fileName = ""; // The file name of the file you want to download.

        try (HttpClient client = HttpClient.newHttpClient()) {
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(downloadUrl + "/file/" + bucketName + "/" + fileName))
                    .header("Authorization", accountAuthorizationToken)
                    .GET()
                    .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            String downloadedData = response.body();
            System.out.println(downloadedData);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
